tests/gradients/test_parameter_shift.py

```diff
@@ -719,6 +723,42 @@ def test_expval_and_variance(self, tol):
         assert gradA == pytest.approx(expected, abs=tol)
         assert gradF == pytest.approx(expected, abs=tol)
 
+    def test_projector_variance(self, tol):
+        """Test that the variance of a projector is correctly returned"""
+        dev = qml.device("default.qubit", wires=2)
+        P = np.array([1])
+        x, y = 0.765, -0.654
+
+        with qml.tape.JacobianTape() as tape:
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.var(qml.Projector(P, wires=0) @ qml.PauliX(1))
+
+        tape.trainable_params = {0, 1}
+
+        res = tape.execute(dev)
+        expected = 0.25 * np.sin(x / 2) ** 2 * (3 + np.cos(2 * y) + 2 * np.cos(x) * np.sin(y) ** 2)
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        # # circuit jacobians

```

Comment:
How about making this its own test case? It would make sure that no state is being left when calling `tape.execute` and would separate the cases better. Costs additional lines though.

Reply:
I would lean to keeping it as is, partly because `dev.execute` should automatically reset between calls, and partly because it matches the structure of the other tests.